tests/rptest/clients/kubectl.py

```diff
@@ -142,15 +142,34 @@ def logger(self) -> Logger:
     def _local_cmd(self, cmd: list[str]):
         """Run the given command locally and return the stdout as bytes.
            Logs stdout and stderr on failure."""
+        def _prepare_output(sout: str, serr: str) -> str:
+            return f"\n--------- stdout -----------\n{sout}" \
+                   f"\n--------- stderr -----------\n{serr}\n"
+
         self._redpanda.logger.info(cmd)
-        try:
-            return subprocess.check_output(cmd, stderr=subprocess.PIPE)
-        except subprocess.CalledProcessError as e:
-            self.logger.info(
-                f'Command failed (rc={e.returncode}).\n' +
-                f'--------- stdout -----------\n{e.stdout.decode()}' +
-                f'--------- stderr -----------\n{e.stderr.decode()}')
-            raise
+
+        # Using text mode to get strings, not binary
+        # Following code will capture all output to log and work
+        # significantly faster than 'check_output'
+        process = subprocess.Popen(cmd,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE,
+                                   text=True)
+        s_out, s_err = process.communicate()
+        self.logger.debug(_prepare_output(s_out, s_err))
+
+        # TODO: Handle s_err output
+        # It will be hard to detect errors as a lot of apps and utils
+        # just uses stderr as normal output. For example, tsh
+
+        process.kill()
+
+        if process.returncode != 0:
+            self.logger.info(f'Command failed (rc={process.returncode}).\n'
+                             f'{_prepare_output(s_out, s_err)}')
+            raise RuntimeError("Last command returned non-zero exit code")
```

Comment:
This is the key exception that will be surfaced in the results, please try to make it as good as the old one, which included the command string, etc. I think you should probably also throw `CalledProcessError` for consistency as that's what we were returning before.

Reply:
Agreed. Will do.